src/main/java/greencity/entities/Category.java

```diff
@@ -0,0 +1,36 @@
+package greencity.entities;
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import javax.persistence.*;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+@Builder
+public class Category {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+
+    @Column(nullable = false, unique = true, length = 100)
+    private String name;
+
+    @ManyToOne
+    private Category category;
+
+    @OneToMany(mappedBy = "category")
+    private List<Category> categories = new ArrayList<>();
```

Comment:
List of categories in Category class?
Some sort of nextlevel pattern :)

Reply:
Same. Is Recursion reference.